sdk/attestation/azure-security-attestation/samples/attestation/attest_sgxenclave_with_runtime_binary.cpp

```diff
@@ -44,7 +44,8 @@ int main()
 
     // create client
     std::string endpoint(GetEnvHelper::GetEnv("ATTESTATION_AAD_URL"));
-    AttestationClient attestationClient(AttestationClient::Create(endpoint));
+    std::shared_ptr<AttestationClient> attestationClient(
+        std::make_unique<AttestationClient>(AttestationClient::Create(endpoint)));
```

Comment:
If I read in a sample, that shared pointer is initialized with unique pointer, I start asking unnecessary questions, why it is written this way, and if there is a reason behind this that I don't see at first glance - and all that drives my attention and time from reading the sample.

I'd either use local variable (most common and simple case), or shared_ptr with make_shared / unique_ptr with make_unique, or even auto with either make_shared or make_unique.

Reply:
You're right, `std::make_shared` makes more sense here.